scripts/globals/mobskills/Petal_Pirouette.lua

```diff
@@ -0,0 +1,23 @@
+---------------------------------------------
+--  Petal Pirouette
+--
+--  Description: Whirling petals reduce tp to zero.
+---------------------------------------------
+
+require("scripts/globals/settings");
+require("scripts/globals/status");
+require("scripts/globals/monstertpmoves");
+
+---------------------------------------------
+function onMobSkillCheck(target,mob,skill)
+    return 0;
+end;
+
+function onMobWeaponSkill(target, mob, skill)
+    local tpReduced = 0;
+    target:setTP(tpReduced);
```

Comment:
Since tpReduced = 0, any reason why you are not just saying `target:setTP(0);` instead?